### PR TITLE
Block import optimizations

### PR DESCRIPTION
**Pull request by arkpar**
Two major optimizations in this PR:
* Moved transaction signature verification to verifier thread
* Exposed codeHash in ExtVM so that evmjit would not need to re-hash it on each execution

### CODE REVIEW

File: libdevcrypto/CryptoPP.cpp [hunk 1]
```diff
@@ -264,7 +264,6 @@ Public Secp256k1::recover(Signature _signature, bytesConstRef _message)
 	
 	ECP::Element x;
 	{
-		Guard l(x_curve);
```
Comment (arkpar) [inline libdevcrypto/CryptoPP.cpp]: This does not seem to be required any longer. I've tested extensively with multithreaded verification - no issues so far.

File: libp2p/RLPxFrameIO.h [hunk 2]
```diff
@@ -56,7 +56,7 @@ class RLPXSocket: public std::enable_shared_from_this<RLPXSocket>
 	
 	bool isConnected() const { return m_socket.is_open(); }
 	void close() { try { boost::system::error_code ec; m_socket.shutdown(bi::tcp::socket::shutdown_both, ec); if (m_socket.is_open()) m_socket.close(); } catch (...){} }
-	bi::tcp::endpoint remoteEndpoint() { try { return m_socket.remote_endpoint(); } catch (...){ return bi::tcp::endpoint(); } }
```
Comment (arkpar) [inline libp2p/RLPxFrameIO.h]: Way too many exceptions thrown here, replaced with non-throwing version.

File: libethereum/BlockChain.cpp [hunk 3]
```diff
@@ -634,24 +635,26 @@ ImportRoute BlockChain::import(BlockInfo const& _bi, bytes const& _block, Overla
...
 		clog(BlockChainDebug) << "Known block just inserted has no details.";
-		clog(BlockChainDebug) << "Block:" << _bi;
+		clog(BlockChainDebug) << "Block:" << _block.info;
 		clog(BlockChainDebug) << "DATABASE CORRUPTION: CRITICAL FAILURE";
 		exit(-1);
 	}
 
 	try {
-		State canary(_db, *this, _bi.hash(), ImportRequirements::DontHave);
+		State canary(_db, *this, _block.info.hash(), ImportRequirements::DontHave);
```
Comment (arkpar) [inline libethereum/BlockChain.cpp]: This additional check impacts performance greatly. It effectively runs all transaction for the second time, redoing signature verification. Is this really needed? I've put it under ETH_DEBUG

**Comment by gavofyork** [comment]
rebase?

**Comment by arkpar** [comment]
rebased

**Comment by gavofyork** [comment]
rebase?

**Pull request merged**